build.rs

```diff
@@ -27,180 +21,46 @@ fn main() {
         }
         Err(env::VarError::NotUnicode(s)) => {
             panic!("VITASDK env var is not a valid unicode but got: {s:?}")
         }
-    }
-
-    let vita_headers_submodule = Utf8Path::new("vita-headers");
-
-    let original_include = vita_headers_submodule.join("include");
-    println!("cargo:rerun-if-changed={original_include}");
+    };
 
-    let out_dir = Utf8PathBuf::from(env::var("OUT_DIR").unwrap());
-    let include = out_dir.join("vita_headers_localized_include");
-
-    localize_bindings(&original_include, &include);
-
-    let headers = Utf8Path::new("src/headers");
-    println!("cargo:rerun-if-changed={headers}");
-    for entry in headers.read_dir_utf8().unwrap() {
-        let entry = entry.unwrap();
-        fs::copy(entry.path(), include.join(entry.file_name())).unwrap();
-    }
-
-    log::info!("Generating preprocessed bindings");
-    let bindings = generate_preprocessed_bindings(&include);
-
-    log::info!("Parsing preprocessed bindings");
-    let mut bindings = syn::parse_file(&bindings).unwrap();
-
-    let db = vita_headers_submodule.join("db");
-
-    log::info!("Loading vita-headers metadata yaml files from \"{db}\"");
-    let mut link = Link::load(db.as_ref(), vitasdk_sys_manifest().as_ref());
-    link.visit_file_mut(&mut bindings);
-
-    let bindings = bindings.into_token_stream();
-
-    let bindings_output = out_dir.join("bindings.rs");
+    // Just to mark it as used
+    #[cfg(not(feature = "bindgen"))]
+    let _ = vitasdk;
 
+    #[cfg(feature = "bindgen")]
     {
-        log::info!("Writing postprocessed bindings into {bindings_output}");
-        let mut bindings_output = io::BufWriter::new(fs::File::create(&bindings_output).unwrap());
-        use std::io::Write;
-        write!(bindings_output, "{bindings}").unwrap();
-    }
-
-    let cargo = env::var_os("CARGO");
-    let mut fmt_cmd = process::Command::new(cargo.as_deref().unwrap_or_else(|| "cargo".as_ref()));
-    fmt_cmd.args(["fmt", "--"]);
-    fmt_cmd.arg(bindings_output);
-
-    log::info!("Running formatting command: {fmt_cmd:?}");
-    let exit_status = fmt_cmd.status().unwrap();
-    if exit_status.success() {
-        log::info!("Formatting command finished");
-    } else {
-        log::warn!("Formatting command failed with status: {exit_status:?}");
-    }
-}
-
-fn generate_preprocessed_bindings(include: &Utf8Path) -> String {
-    bindgen::Builder::default()
-        .header(include.join("all.h"))
-        .clang_args(&["-target", "armv7a-none-eabihf"])
-        .parse_callbacks(Box::new(BindgenCallbacks::new()))
-        .use_core()
-        .ctypes_prefix("crate::ctypes")
-        .generate_comments(false)
-        .prepend_enum_name(false)
-        .formatter(bindgen::Formatter::None)
-        .generate()
-        .expect("Bindgen failed")
-        .to_string()
-}
-
-// Replace `#include <>` with `#include ""`
-fn localize_bindings(original_include: &Utf8Path, localized_include: &Utf8Path) {
```

Comment:
Why have you removed that?

Reply:
AFAIK there's no advantage in manually localizing bindings. `-I` should have basically the same behavior without requiring manually patching all upstream headers. See https://clang.llvm.org/docs/ClangCommandLineReference.html#include-path-management and https://gcc.gnu.org/onlinedocs/gcc/Directory-Options.html

Reply:
Huh. Didn't think about that. I'll test that.